resources/configure.rb

```diff
@@ -0,0 +1,93 @@
+
+#
+# Cookbook:: sql_server
+# Resource:: install
+#
+# Copyright:: 2017, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+property :reg_version, String
+property :version, String, default: '2012'
+property :tcp_enabled, [true, false], default: true
+property :sql_port, Integer, default: 1433
+property :tcp_dynamic_ports, String, default: ''
+property :np_enabled, [true, false], default: false
+property :sm_enabled, [true, false], default: true
+property :via_default_port, String, default: '0:1433'
+property :via_enabled, [true, false], default: false
+property :via_listen_info, String, default: '0:1433'
+property :agent_startup, String, equal_to: ['Automatic', 'Manual', 'Disabled', 'Automatic (Delayed Start)'], default: 'Disabled'
+
+action :service do
+  # Compute service name based on sql server instance name defined as the resource name
+  service_name = (new_resource.name != 'MSSQLSERVER') ? "MSSQL$#{new_resource.name}" : new_resource.name
+
+  # Agent name needs to be declared because if you use the SQL Agent, you need
+  # to restart both services as the Agent is dependent on the SQL Service
+  agent_service_name = (new_resource.name == 'MSSQLSERVER') ? 'SQLSERVERAGENT' : "SQLAgent$#{new_resource.name}"
+
+  # Compute registry version based on sql server version
+  reg_version = new_resource.reg_version || ::SqlServer::Helper.reg_version_string(new_resource.version)
+
+  reg_prefix = "HKLM\\SOFTWARE\\Microsoft\\Microsoft SQL Server\\#{reg_version}#{new_resource.name}\\MSSQLServer"
+
+  # Configure Tcp settings - static tcp ports
+  registry_key "#{reg_prefix}\\SuperSocketNetLib\\Tcp\\IPAll" do
+    values [{ name: 'Enabled', type: :dword, data: new_resource.tcp_enabled ? 1 : 0 },
+            { name: 'TcpPort', type: :string, data: new_resource.sql_port.to_s },
+            { name: 'TcpDynamicPorts', type: :string, data: new_resource.tcp_dynamic_ports.to_s }]
+    recursive true
+    notifies :restart, "service[#{service_name}]", :immediately
```

Comment:
Do we really want immediately on the restart here? This could potentially restart sqlserver 4 times on the first install. Delayed is delayed for the entire resource so it would still restart at the end of the resource converge which is probably acceptable.